src/generic_ui/scripts/model.ts

```diff
@@ -104,12 +104,29 @@ export class Model {
     return null;
   }
 
-  public updateGlobalSettings = (settings :Object) => {
+  public removeContact = (user :User) : void => {
+    var categories = user.getCategories();
```

Comment:
I think you can simplify this a bit by using the categorizeUser method and pass null as newCategory (similar to whats done in removeNetwork).  So it could probably just be something like
```
var categories = user.getCategories();
categorizeUser(user, this.contacts.getAccessContacts, categories.getTab, null);
// similar call to categorizeUser(..) but for share contacts
```